mesonbuild/backend/xcodebackend.py

```diff
@@ -720,6 +739,47 @@ def generate_pbx_build_style(self, objects_dict: PbxDict) -> None:
             settings_dict.add_item('COPY_PHASE_STRIP', 'NO')
             styledict.add_item('name', f'"{name}"')
 
+    def to_shell_script(self, args: CompilerArgs) -> str:
+        quoted_cmd = []
+        for c in args:
+            quoted_cmd.append(c.replace('"', chr(92) + '"'))
+        return f"\"#!/bin/bash\\n{' '.join(quoted_cmd)}\\n\""
+
+    def generate_pbx_build_rule(self, objects_dict: PbxDict) -> None:
+        for name, languages in self.build_rules.items():
+            target: BuildTarget = self.build_targets[name]
+            for language, idval in languages.items():
+                compiler: Compiler = target.compilers[language]
+                buildrule = PbxDict()
+                buildrule.add_item('isa', 'PBXBuildRule')
+                buildrule.add_item('compilerSpec', 'com.apple.compilers.proxy.script')
+                if compiler.get_id() != 'yasm':
+                    # Yasm doesn't generate escaped build rules
+                    buildrule.add_item('dependencyFile', '"$(DERIVED_FILE_DIR)/$(INPUT_FILE_BASE).d"')
+                buildrule.add_item('fileType', NEEDS_CUSTOM_RULES[language])
+                inputfiles = PbxArray()
+                buildrule.add_item('inputFiles', inputfiles)
+                buildrule.add_item('isEditable', '0')
+                outputfiles = PbxArray()
+                outputfiles.add_item('"$(DERIVED_FILE_DIR)/$(INPUT_FILE_BASE).o"')
+                buildrule.add_item('outputFiles', outputfiles)
+                # Do NOT use this parameter. Xcode will accept it from the UI,
+                # but the parser will break down inconsistently upon next
+                # opening. rdar://FB12144055
+                # outputargs = PbxArray()
+                # args = self.generate_basic_compiler_args(target, compiler)
+                # outputargs.add_item(self.to_shell_script(args))
+                # buildrule.add_item('outputFilesCompilerFlags', outputargs)
+                commands = CompilerArgs(compiler)
+                commands += compiler.get_exelist()
+                if compiler.get_id() == 'yasm':
+                    # Yasm doesn't generate escaped build rules
+                    commands += self.compiler_to_generator_args(target, compiler, output='"$SCRIPT_OUTPUT_FILE_0"', input='"$SCRIPT_INPUT_FILE"', depfile=None)
+                else:
+                    commands += self.compiler_to_generator_args(target, compiler, output='"$SCRIPT_OUTPUT_FILE_0"', input='"$SCRIPT_INPUT_FILE"', depfile='"$(dirname "$SCRIPT_OUTPUT_FILE_0")/$(basename "$SCRIPT_OUTPUT_FILE_0" .o).d"', extras=['$OTHER_INPUT_FILE_FLAGS'])
+                buildrule.add_item('script', self.to_shell_script(commands))
```

Comment:
Long lines like these should be formatted to have one function argument per line lined up to the opening brace.